docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -134,6 +135,131 @@ function template_preprocess_uiowa_core_headline(&$variables) {
 
 }
 
+/**
+ * Implements hook_preprocess_media().
+ */
+function uiowa_core_preprocess_media(&$variables) {
+
+  // Detect if this is a figure and if it is, it has already been processed for inline sizing and alignment.
+  $is_figure = FALSE;
+  if (isset($variables['attributes']['data-figure'])) {
+    $is_figure = filter_var($variables['attributes']['data-figure'], FILTER_VALIDATE_BOOLEAN);
+  }
+
+  // Check to see if the media element has been processed, or if it is an embed.
+  if (!$is_figure && isset($variables['elements']['#embed']) && $variables['elements']['#embed']) {
+
+    // If we have a view mode set...
+    if (isset($variables['view_mode'])) {
+      $view_mode = $variables['view_mode'];
+
+      // Process the view mode classes.
+      $view_mode_modified = uiowa_core_inline_viewmode_map($view_mode);
+
+      // If we have relevant view mode classes...
+      if (!empty($view_mode_modified)) {
+
+        // Attach the inline sizing library and assign the classes.
+        $variables['#attached']['library'][] = $view_mode_modified['library'];
+        $variables['attributes']['class'][] = $view_mode_modified['class'];
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_filter_caption().
+ */
+function uiowa_core_preprocess_filter_caption(&$variables) {
+  // Get the '<drupal-media>' tag string.
+  $drupal_media_markup = strval($variables['node']);
+
+  // Define some $replacement_strings to remove.
+  $replacement_strings = ['<drupal-media ', '></drupal-media>'];
+
+  // Remove the $replacement_strings.
+  $drupal_media_attributes_string = str_replace($replacement_strings, '', $drupal_media_markup);
+
+  // Separate the attributes of the tag string in to an array of attribute strings.
+  // Ex: ['foo="bar"', 'ban="baz"'].
+  $drupal_media_attributes_strings = explode(' ', $drupal_media_attributes_string);
+
+  // Create a new array to be used later for the associative media attributes data.
+  $associative_drupal_media_attributes_strings = [];
+
+  // For each of the media attributes strings...
+  foreach ($drupal_media_attributes_strings as $value) {
+
+    // Separate them in to two parts, clean them up, and add them to the associative array.
+    $attribute__value = explode('=', $value);
+    $associative_drupal_media_attributes_strings[$attribute__value[0]] = str_replace('"', '', $attribute__value[1]);
+  }
```

Comment:
At first I was thinking we could use an HTML parser to pull out the attributes for this, but I think that's over the top, needing to convert things back and forth.

But I was wondering if we could do something like
```
  // Retreive the drupal media attributes.
  preg_match_all('%\s?([^"]*)="([^"]*)"%is', $drupal_media_attributes_string, $matches);
  $associative_drupal_media_attributes_strings = [];
  for ($i = 0; $i < count($matches[0]); $i++) {
    $associative_drupal_media_attributes_strings[$matches[1][$i]] = $matches[2][$i];
  }
  ```
  The preg_match there grabs pairs of not-in-quotes and in-quotes values (and drops the leading space if it's there), and then a similar pairing up to make the associative array.

I'd guess the preg_match_all would be technically faster since it can do the search and cleanup in one go, but am open to preferences because 1) the strings that'll be parsed here shouldn't be long enough for processing to be that intense and 2) I'll admit that regexing everything doesn't always leave the most readable code for future maintenance.

Reply:
```suggestion
  // Get the '<drupal-media>' tag string.
  $drupal_media_markup = strval($variables['node']);
  
  // Define some $replacement_strings to remove.
  $replacement_strings = ['<drupal-media ', '></drupal-media>'];

  // Remove the $replacement_strings.
  $drupal_media_attributes_string = str_replace($replacement_strings, '', $drupal_media_markup);

  // Retreive the drupal media attributes and separate
  // them into an array of attribute strings.
  // Ex: ['foo="bar"', 'ban="baz"'].
  preg_match_all('%\s?([^"]*)="([^"]*)%is', $drupal_media_markup, $matches);
  $associative_drupal_media_attributes_strings = [];
  for ($i = 0; $i < count($matches[0]); $i++) {
    $associative_drupal_media_attributes_strings[$matches[1][$i]] = $matches[2][$i];
  }
```

Reply:
What does `$variables['node']` contain here? I would usually expect this to be the node object (although it would usually be listed as `$variables['#node'] in that case).

Reply:
While this is well-commented in terms of explaining what is happening from a code perspective, it does not provide any context as to what is happening generally or why. Despite the comments, if I were reading this, I would have no idea why this piece of code existed if I was later evaluating whether it was still needed (for example, when we upgrade to CKEditor 5 and have to revamp all our WYSIWYG code).

Reply:
> What does $variables['node'] contain here? I would usually expect this to be the node object (although it would usually be listed as `$variables['#node'] in that case).

It has the media entity, like `<drupal-media data-entity-type="media" data-entity-uuid="ffc45612-f7e6-46c3-b9e2-17ce7861be49" data-view-mode="medium__no_crop"></drupal-media>`. I'm not sure if there are times when it has other things--so far it's been the only thing I've seen come through it.